app/helpers/enroller_query_helper.rb

```diff
@@ -0,0 +1,158 @@
+      raise InvalidQueryError.new(:jurisdiction, jurisdiction)
+    end
+
+    # Validate jurisdiction scope
+    scope = query[:scope]
+    raise InvalidQueryError.new(:scope, scope) unless scope.nil? || %w[all exact].include?(scope)
+
+    # Validate assigned user
+    user = query[:user]
+    raise InvalidQueryError.new(:user, user) unless user.nil? || %w[none].include?(user) || user.to_i.between?(1, 999_999)
+
+    # Validate sorting order
+    order = query[:order]
+    raise InvalidQueryError.new(:order, order) unless order.nil? || order.blank? || %w[name jurisdiction assigned_user state_local_id sex dob
+                                                                                       enrollment_date].include?(order)
+
+    # Validate sorting direction
+    direction = query[:direction]
+    raise InvalidQueryError.new(:direction, direction) unless direction.blank? || %w[asc desc].include?(direction)
+    raise InvalidQueryError.new(:direction, direction) unless (order.present? && direction.present?) || (order.blank? && direction.blank?)
+
+    query
+  end
+
+  def patients_by_query(current_user, query)
+    # Determine jurisdiction
+    jurisdiction = Jurisdiction.find(query[:jurisdiction].to_i) unless ['all', nil].include?(query[:jurisdiction])
+    jurisdiction = current_user.jurisdiction if jurisdiction.nil?
+
+    # Get enrolled patients by current user
+    patients = current_user.enrolled_patients
+
+    # Filter by assigned jurisdiction
+    patients = patients.where(jurisdiction_id: jurisdiction.subtree_ids)
+
+    # Filter by scope
+    patients = patients.where(jurisdiction_id: jurisdiction.id) if query[:scope] == 'exact'
+
+    # Filter by assigned user
+    patients = patients.where(assigned_user: query[:user] == 'none' ? nil : query[:user].to_i) unless query[:user].nil?
+
+    # Filter by search text
+    patients = filter_by_text(patients, query[:search])
+
+    # Sort
+    sort(patients, query[:order], query[:direction])
+  end
+
+  def filter_by_text(patients, search)
+    return patients if search.nil? || search.blank?
+
+    filtered = patients.where('first_name like ?', "#{search&.downcase}%").or(
+      patients.where('last_name like ?', "#{search&.downcase}%").or(
+        patients.where('user_defined_id_statelocal like ?', "#{search&.downcase}%").or(
+          patients.where('user_defined_id_cdc like ?', "#{search&.downcase}%").or(
+            patients.where('user_defined_id_nndss like ?', "#{search&.downcase}%").or(
+              patients.where('date_of_birth like ?', "#{search&.downcase}%").or(
+                patients.where('patients.email like ?', "#{search&.downcase}%")
+              )
+            )
+          )
+        )
+      )
+    )
+
+    phone_query = search.delete('^0-9')
+    phone_query.blank? ? filtered : filtered.or(patients.where('primary_telephone like ?', "+1#{phone_query}%"))
```

Comment:
The reason this was added because of the shared search bar here: https://github.com/SaraAlert/SaraAlert/blob/960-enroller-table-refactor/app/javascript/components/patient/PatientsTable.js#L705.  So either the tooltip needs to be configurable OR we support searching by phone number on all `PatientsTable` instances.  I'm okay with either so up to you or @hackrm

Reply:
Ah right, I see. Ok I think this is fine as is then. The pub health dashboard also allows search for fields not shown in the table (CDC ID, NNDSS IDD, email).